pilot/pkg/features/experimental.go

```diff
@@ -180,6 +180,17 @@ var (
 		false,
 		"If enabled, controllers required for ambient will run. This is required to run ambient mesh.").Get()
 
+	EnableAmbientWaypoints = func() bool {
```

Comment:
This is to make ambient conflex to understand,  the ideal way is: we have a flag indicating whether to enable ambient or not. 
I could not understand it even reading this description.

Reply:
The user experience for ambient does not change: --set profile=ambient.

the purpose of this change is to make it so we can default enable a subset if feature's.

The current plan is to promote ztunnel to beta in 1.22. Beta means we need the controller on by default. Waypoints are alpha so they are inappropriate to turn on by default.

Reply:
waypoint is the most important component of ambient, without it how can we call beta

Reply:
I disagree it being the "most important". Important, yes, but not *the* most important. Not everyone needs/wants L7 processing.

Reply:
Oh, no. This breaks my knowledge. From the begining, L7 is high lighted. https://istio.io/v1.15/blog/2022/introducing-ambient-mesh/

can users imagine a mesh does not support L7 features

Reply:
Right, highlighted.. but it's not *the most* important. It is one important component that makes up ambient, but it isn't more important than say the CNI plugin or ztunnel. Delaying L4/ztunnel moving to beta for L7/waypoint doesn't make sense, because they are different components. There are some users that don't necessary want/need the L7.

Reply:
It is good to announce waypoint is alpha. And it equals ambient l4 beta  but not ambient beta

Reply:
> I disagree it being the "most important". Important, yes, but not _the_ most important. Not everyone needs/wants L7 processing.

that the needs of users in different locations may vary.
Many users in China maybe not need ztunnel, while waypoint L7 is essential.

Because most users in China do not have a need for FIPS. And some of the features of L4 can be adopted from Kuberntes CNI, such as NetworkPolicy, although it is not as good as AuthoriztionPolicy, but for many users, it is enough.

I'd like to propose a PR to see if it's possible to not enable most of ztunnel's features, but only enable DNS proxy, and use dnx proxy forward the traffic directly to waypoint.